src/test/java/com/vackosar/gitflowincrementalbuild/integration/MavenIntegrationTestBase.java

```diff
@@ -338,11 +348,35 @@ public void buildWithSingleSelectedModule_alsoMakeDependends() throws Exception
                 .doesNotContain(" testJarDependent");
     }
 
-    private void checkoutDevelop() throws GitAPIException, CheckoutConflictException, RefAlreadyExistsException,
+    @Test
+    public void quarkusScenario_noChanges() throws Exception {
+        final String output = executeBuild(quarkusScenarioProps());
+
+        assertThat(output).contains("Executing validate goal on current project only")
+                .contains(" parent")
+                .doesNotContain(" bom")
+                .doesNotContain(" child1")
+                .doesNotContain(" child2");
+    }
+
+    @Test
+    public void quarkusScenario_bomChanged() throws Exception {
+        Files.write(repoPath.resolve("quarkus-scenario").resolve("bom").resolve("pom.xml"), Arrays.asList("<!-- changed -->"), StandardOpenOption.APPEND);
+
+        final String output = executeBuild(quarkusScenarioProps());
+
+        assertThat(output)
+                .doesNotContain(" parent")
+                .contains(" bom")
+                .doesNotContain(" child1")
+                .contains(" child2");
+    }
+
+    private void checkout(Branch branch) throws GitAPIException, CheckoutConflictException, RefAlreadyExistsException,
             RefNotFoundException, InvalidRefNameException {
         Git git = localRepoMock.getGit();
         git.reset().setMode(ResetCommand.ResetType.HARD).setRef("HEAD").call();
-        git.checkout().setName("develop").call();
+        git.checkout().setName(Branch.QUARKUS.branchName).call();
```

Comment:
Shouldn't this checkout the branch given in parameter `branch`?

Reply:
Oh, good catch!